addon-test-support/@ember/test-helpers/dom/type-in.js

```diff
@@ -0,0 +1,98 @@
+import { nextTickPromise } from '../-utils';
+import settled from '../settled';
+import getElement from './-get-element';
+import isFormControl from './-is-form-control';
+import { __focus__ } from './focus';
+import { Promise } from 'rsvp';
+import fireEvent from './fire-event';
+
+/**
+ * Mimics character by character entry into the target `input` or `textarea` element.
+ *
+ * Allows for simulation of slow entry by passing an optional millisecond delay
+ * between key events.
+
+ * The major difference between `typeIn` and `fillIn` is that `typeIn` triggers
+ * keyboard events as well as `input` and `change`.
+ * Typically this looks like `focus` -> `focusin` -> `keydown` -> `keypress` -> `keyup` -> `input` -> `change`
+ * per character of the passed text (this may vary on some browsers).
+ *
+ * @public
+ * @param {string|Element} target the element or selector to enter text into
+ * @param {string} text the test to fill the element with
+ * @param {number} millisecondDelay number of millisecons to wait per keypress
+ * @return {Promise<void>} resolves when the application is settled
+ */
+export default function typeIn(target, text, millisecondDelay = 0) {
```

Comment:
Lets go with an `{ delay: 50 }` for the third arg

Reply:
done